src/lib/storage/storage_manager.cpp

```diff
@@ -260,4 +276,284 @@ std::ostream& operator<<(std::ostream& stream, const StorageManager& storage_man
+                                         const std::vector<uint32_t>& segment_offset_ends, const std::string& file_name) {
+  chunk_header header;
+  header.row_count = chunk->size();
+  header.segment_offset_ends = segment_offset_ends;
+
+  export_value(header.row_count, file_name);
+
+  for (const auto segment_offset_end : header.segment_offset_ends) {
+    export_value(segment_offset_end, file_name);
+  }
+
+  const auto segment_count = chunk->column_count();
+  for (auto segment_index = size_t{0}; segment_index < segment_count; ++segment_index) {
+    const auto abstract_segment = chunk->get_segment(static_cast<ColumnID>(static_cast<uint16_t>(segment_index)));
+    const auto dict_segment = dynamic_pointer_cast<DictionarySegment<int>>(abstract_segment);
+
+    write_dict_segment_to_disk(dict_segment, file_name);
+  }
+}
+
+void StorageManager::persist_chunks_to_disk(const std::vector<std::shared_ptr<Chunk>>& chunks, const std::string& file_name) {
+  /*
+    TODO(everyone): Think about a proper implementation of a locking method. Each written file needs to be
+    locked prior to writing (and released afterwards). It was decided to use the mutex class by cpp.
+    (see https://en.cppreference.com/w/cpp/thread/mutex).
+  */
+  // file_lock.acquire();
+
+  auto chunk_segment_offset_ends = std::vector<std::vector<uint32_t>>(StorageManager::_chunk_count);
+  auto chunk_offset_ends = std::array<uint32_t, StorageManager::_chunk_count>();
+  auto chunk_ids = std::array<uint32_t, StorageManager::_chunk_count>();
+
+  auto offset = uint32_t{sizeof(file_header)};
+  for (auto chunk_index = uint32_t{0}; chunk_index < chunks.size(); ++chunk_index) {
+    const auto segment_offset_ends = generate_segment_offset_ends(chunks[chunk_index]);
+    offset += segment_offset_ends.back();
+
+    chunk_segment_offset_ends[chunk_index] = segment_offset_ends;
+    chunk_offset_ends[chunk_index] = offset;
+  }
+  // Fill all offset fields, that are not used with 0s.
+  for (auto chunk_index = chunks.size(); chunk_index < StorageManager::_chunk_count; ++chunk_index) {
+    chunk_offset_ends[chunk_index] = uint32_t{0};
+  }
+
+  // TODO(everyone): Find, how to get the actual chunk id.
+  for (auto index = uint32_t{0}; index < StorageManager::_chunk_count; ++index) {
+    chunk_ids[index] = index;
+  }
+
+  file_header fh;
+  fh.storage_format_version_id = _storage_format_version_id;
+  fh.chunk_count = static_cast<uint32_t>(chunks.size());
+  fh.chunk_ids = chunk_ids;
+  fh.chunk_offset_ends = chunk_offset_ends;
+
+  export_value<file_header>(fh, file_name);
+
+  for (auto chunk_index = uint32_t{0}; chunk_index < chunks.size(); ++chunk_index) {
+    const auto chunk = chunks[chunk_index];
+    write_chunk_to_disk(chunk, chunk_segment_offset_ends[chunk_index], file_name);
+  }
+
+  // file_lock.release();
+}
+
+file_header StorageManager::read_file_header(const std::string& filename) {
+  file_header file_header;
```

Comment:
We tend to be rather strict with `auto to stick`. Please try to use it as much as possible.

Reply:
will be fixed in next commit :)